dawgz/workflow.py

```diff
@@ -264,6 +261,20 @@ def prune(*jobs) -> List[Job]:
             if len(pending) < len(job.array):
                 job.array = pending
 
+        done = {
+            dep for dep, status in job.dependencies.items()
+            if dep.done() and status == 'failure'
+        }
+
+        if done:
+            job.detach(*done)
+            if len(job.dependencies) == 0:
+                if job in _jobs:
+                    _jobs.remove(job)
+                for j in dfs(job):  # Remove failed branch
+                    if j in _jobs:
+                        _jobs.remove(j)
```

Comment:
This means that dependency fails are silent. This is not acceptable.

Reply:
It is not a dependency whenever the postconditions have been satisfied (the failure will never be executed). So this is expected behavior.

Reply:
Hum so what you mean is that, if at prune-time something is sure to have its unsatisfied dependencies. It should be pruned silently?